### DISCUSSION TITLE
Nested object in useForm hook

### DISCUSSION THREAD

**Discussion by danielbettech** [mantinedev/mantine#1697] [Q&A] [answered]
For organizational purposes I would like to set initialValues in the useForm hook with nested object so I don't have to map the values later.

example:

```
const form = useForm({
  initialValues: {
    firstName: '',
    lastName: '',
    age: '',
    address: {
      addressLineOne: '',
      addressLineTwo: '',
      zip: '',
      country: '',
    },
  },
})
```

Is this possible?

**Comment by rtivital** [comment, marked as answer]
It will be supported from version 5.0 – https://v5.mantine.dev/form/nested/
Versions below 5.0 do not support that. If you need this feature now, you can update the package to 5.0 alpha version.

**Reply by danielbettech** [reply]
Thanks!
